### ISSUE TITLE
Kubernetes PathPrefixStrip

### ISSUE THREAD

**Issue by ekozan** [containous/traefik#337]
The kubernetes provider create the path rules with PathPrefixStrip so the service fail 

example : 
``` 
Route	               Rule
/                      PathPrefixStrip:/
grafana.domain.tdl	   Host:grafana.domain.tdl
```

browser http://grafana.domain.tdl/public/css/grafana.dark.min.a95b3754.css -> to the service http://grafana.domain.tdlpublic/css/grafana.dark.min.a95b3754.css

I think we need to put Path for this usage

**Comment by emilevauge** [comment]
@ekozan why are you declaring a `/` path in your ingress ?
Why not simply:

```
spec:
  rules:
  - host: grafana.domain.tdl 
    http:
      paths:
      - backend:
          serviceName: grafana
          servicePort: 80
```

**Comment by ekozan** [comment]
Hum ok but what if 
admin.domain.tdl/.well-known/  my acme serveur wait for  admin.domain.tdl/.well-known/acme  and not for  admin.domain.tdl/acme

for me the default behavior is Path and we need to add the possibility in some case to switch to PathPrefixStrip with label  maybe labels can help for http https problems too

**Comment by emilevauge** [comment]
@ekozan I can change the default behavior to `PathPrefix` but I'm sure that someone will soon open an issue because he was not expecting that ;)
Indeed, I could use services labels like what is done in Docker/Marathon backends.

**Comment by ekozan** [comment]
I just need to have the posibility of changing this. 

on official ingress controller it's work like PathPrefix

**Comment by emilevauge** [comment]
OK, right. I can change the default behavior.
But in your example, if your ACME server wait for `admin.domain.tdl/.well-known/acme`, your can simply use a Host rule `admin.domain.tdl` and it will send all your requests keeping path.

**Comment by ekozan** [comment]
```
admin.domain.tdl/.well-known/acme -> Let'sEncrypt Pod
admin.domain.tdl                              -> Admin Pod
```

Maybe now with traefik I can create one regex rules I have just try a switch without any modification from the official ingress

**Comment by ekozan** [comment]
ping @mgoodness

**Comment by mgoodness** [comment]
I'd like to see Traefik default to the same behavior as the nginx controller. Given this IngressSpec:
```
spec:
  rules:
    - host: admin.domain.tld
      http:
        paths:
          - path: /
            backend:
              serviceName: service1
              servicePort: 80
          - path: /bar
            backend:
              serviceName: service2
              servicePort: 8080
          - path: /baz/test/
            backend:
              serviceName: service3
              servicePort: 80
```
I would expect:
- `http://admin.domain.tld/foo.css`, `http://admin.domain.tld/foo/bar/`, and `http://admin.domain.tld/baz/foo.html` to be routed to `service1:80` without the path being rewritten
- `http://admin.domain.tld/bar.js` and `http://admin.domain.tld/bar/baz` to be routed to `service2:8080` without rewrite
- `http://admin.domain.tld/baz/test/foo to be routed to `service3:80` without rewrite

I believe that would mean making `PathPrefix` the default behavior. Overriding it could be done by reading Service or Ingress annotations (I think those are more appropriate to this use than labels).

Another idea is to hard-code the default rules into the `kubernetes.tmpl` file and only template the Hosts and Paths. If someone wants to override, they can pass a new template via ConfigMap. I've been toying with that in my own branch.

**Comment by mgoodness** [comment]
Looks like the behavior I outlined above is dependent on #298

**Comment by emilevauge** [comment]
@mgoodness not really, I will just change the rule to `PathPrefix` it will be compliant with the nginx controller ;)

**Comment by mgoodness** [comment]
@emilevauge I made that change on my branch, but it results in the `/` route matching everything.

**Comment by emilevauge** [comment]
Ouch, I didn't understand that you were talking about that point.
There is an issue with gorillatoolkit.org/pkg/mux and for now, I don't really know how to deal with it :'(

**Comment by mgoodness** [comment]
I understand. Probably out of scope for this issue anyway. I'll see if I can help with that, and/or some of the other things I've mentioned 😄

**Comment by ekozan** [comment]
O.o" so in my case 

```
admin.domain.tdl/.well-known/acme -> Let'sEncrypt App
admin.domain.tdl                  -> Admin App
```

it'll always go to  Admin App ?

**Comment by emilevauge** [comment]
It depends of the order in fact.
